### DISCUSSION TITLE
[Question]: What are these `english` and `translated` keys used in l10n files?

### DISCUSSION THREAD

**Discussion by KiGamji** [danny-avila/LibreChat#3134] [Q&A] [answered]
### What is your question?

When I decided to improve Russian localization, I noticed that all language files have the strings doubled with `english` and `translated` keys:
![image](https://github.com/danny-avila/LibreChat/assets/59279014/7fc0a9ed-de79-4215-b6f0-2f8cc22d2fe9)
![image](https://github.com/danny-avila/LibreChat/assets/59279014/c0dd0772-31c5-4e7f-92ef-df5ec4ad63a9)

With that being said, they are not present in the English localization file:
![image](https://github.com/danny-avila/LibreChat/assets/59279014/7bacc82b-974d-48a3-8b8e-317249e79a50)

Also, the position of strings is different on non-English localization files and english (or some of them are missing?):
![image](https://github.com/danny-avila/LibreChat/assets/59279014/a6aeb6a0-b026-426e-a2e0-0456460db321)
![image](https://github.com/danny-avila/LibreChat/assets/59279014/cd9bdea0-ed77-4926-b406-5f780019c5fc)

I'm now confused. Why are the strings doubled? What should I do with missing/order mismatching strings? Should I reorder them to match the original order?

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment, marked as answer]
thanks for looking into this.

The order is not important. It would be nice to have them organized alphabetically by the key, but not a requirement.

The keys showing both the English and other language is just for comparison, to see if they need improvement. They are not important otherwise as they are not used by the project, it’s mainly to guide for the translator.

these are automatically generated so I can add a note to the script with these details. It’s possible they may be out of date as the script is currently ran manually.

**Reply by KiGamji** [reply]
So I can drop the keys, right? If so, I'll take the Eng.ts file as a base to translate from scratch.

**Reply by KiGamji** [reply]
@danny-avila

**Reply by danny-avila** [reply]
> So I can drop the keys, right? If so, I'll take the Eng.ts file as a base to translate from scratch.

I'm not sure which ones you mean. if the bottom ones with the English comparisons, yes, they can be removed. but they will also be re-generated soon whenever I run the script, so I would just leave them as is.
